sudospawner/spawner.py

```diff
@@ -94,6 +94,19 @@ def start(self):
 
     @gen.coroutine
     def _signal(self, sig):
+        if sig == 0:
+            # short-circuit existence check without invoking sudo
+            try:
+                os.kill(self.pid, sig)
+            except ProcessLookupError:
+                # No such process
+                return False
+            except PermissionError:
+                # When running hub with reduced permissions,
+                # we won't have permission to send signals, even 0.
+                # PermissionError means process exists.
```

Comment:
The process might exist, but it might not be the notebook process as we found in #33. What impact will that have on hub behavior?